src/main/java/com/nextcloud/common/OkHttpMethodBase.kt

```diff
@@ -39,7 +39,7 @@ import okhttp3.Response
  */
 abstract class OkHttpMethodBase(var uri: String,
                                 val useOcsApiRequestHeader: Boolean) {
-    lateinit var response: Response
+    private var response: Response? = null
     var queryMap: Map<String, String> = HashMap()
```

Comment:
@tobiasKaminsky Is there a reason why those members are public?

Reply:
No, I just forgot that the default visibility in Kotlin is different than in Java.

Reply:
I wanted to avoid null, and thus I used lateinit, but seems to be wrong?

Reply:
Yes, it is incorrect use. Keyword latetinit is a workaround for situations where you want to declare a non-nullable value, but it cannot be initialized immediately. Kotlin requires all variables to be initailized in a primary constructor (this eliminates whole class of bugs). When this is not possible (for example, when you can set it in `onCreate()`, you can annotate a variable `lateinit`. Kotlin will generate some code that checks if variable is initialized before first use and throw NPE if not, but the variable is still non-nullable.

This is not an alternative for `java.util.Optional` or nullable type.